[WIP] Move unit tests which were in integration tests - More py3 and mocking fixes

s0undt3ch opened the pull request:
This needs to be merged after https://github.com/saltstack/salt/pull/40138 is merged

s0undt3ch committed:
Disable pip wheel usage for this test.

The pip cache dir will be owned by root and not the user created.
Revisit this issue ASAP.

s0undt3ch committed:
We actually need to fix the pip state to recognize new/deprecated options

s0undt3ch committed:
Make the option properties available in the config mixin instead

s0undt3ch committed:
Transplant configurations right after CLI parsing is complete

s0undt3ch committed:
These are not integration tests

salt-jenkins commented:
@s0undt3ch, thanks for your PR! By analyzing the history of the files in this pull request, we identified @cachedout, @castor-morveer and @opdude to be potential reviewers.

s0undt3ch committed:
Py3 compat

s0undt3ch committed:
Fix pytest loading if a system wide salt package is installed

s0undt3ch committed:
Reliable sorting on both python 2 and 3

s0undt3ch committed:
Rename decorators in order not to be picked up by pytest as test cases

s0undt3ch committed:
Update tests to the new sorting

s0undt3ch committed:
Code style fixes

s0undt3ch committed:
Skip tests on required missing libs

s0undt3ch committed:
This is not an integration test. Skip test.

Skipping tests since I have no clue what is supposed to be tested here
with that `raise Exception(ret)` at the end.

s0undt3ch committed:
Don't run these tests under Py3. The PSF takes care of that.

cachedout reviewed (changes requested):
Just one lint error: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/9957/violations/file/tests/unit/utils/test_docker.py/

s0undt3ch committed:
Py3 compat

s0undt3ch committed:
Use random ports. Tests cleanup.

s0undt3ch committed:
Add file to ignored paths

s0undt3ch committed:
Take into account the comment from @damon-atkins. Improved MS Win support.

s0undt3ch committed:
Proper mocking.

s0undt3ch committed:
Pylint disable

s0undt3ch committed:
Why do need the system grains here?! This is an expensive call.

s0undt3ch committed:
Use tests.support.unit instead

s0undt3ch committed:
Minor logic fix

s0undt3ch committed:
Proper mocking

s0undt3ch committed:
This just feels wrong

s0undt3ch committed:
Style cleanup

s0undt3ch committed:
We now sync all on minion connect

s0undt3ch committed:
We need order, specially under python 3

s0undt3ch commented:
Go Go Jenkins!

s0undt3ch commented:
Go Go Jenkins!

s0undt3ch committed:
Move tests to the right test module

s0undt3ch committed:
Py3 compat

s0undt3ch committed:
Lint

s0undt3ch committed:
Take care of some more deprecated TestCase methods

s0undt3ch committed:
Stop using deprecated TestCase methods

s0undt3ch committed:
Minor lint fix

s0undt3ch committed:
Proper mocking.

s0undt3ch committed:
Error message in a single line

s0undt3ch committed:
Proper mocking.

s0undt3ch committed:
Add mixin to assert XML equality

s0undt3ch committed:
Proper mocking.

s0undt3ch committed:
Py3 Compat

s0undt3ch committed:
Proper mocking.

s0undt3ch committed:
Py3 compat. Cast to list which works in Py2 and Py3.

s0undt3ch committed:
Proper mocking.

s0undt3ch committed:
Misplaced test. Use tests.support.unit instead.

s0undt3ch commented:
@cachedout this should be ready for you to take another look

s0undt3ch commented on tests/unit/modules/test_file.py:
```diff
@@ -759,17 +761,89 @@ def setup_loader_modules(self):
+                                           'cp.list_master_dirs': MagicMock(return_value=[])}):
+            ret = filemod.source_list(
+                [{'file://' + self.myfile: ''}], 'filehash', 'base')
+            print(8888, ret)
```
Bah!

s0undt3ch commented on tests/unit/netapi/rest_tornado/test_handlers.py:
```diff
@@ -3,27 +3,28 @@
+from tests.support.unit import TestCase, skipIf
 
-# Import Salt libs
+# Import Salt libse
```
Fix comment

s0undt3ch commented:
Fixed my own review comments.

s0undt3ch committed:
Use a temporary configuration

s0undt3ch committed:
This argument is not required when used as a ctx manager

s0undt3ch committed:
Allow creating a temporary configuration to be used for a single test

s0undt3ch committed:
Use a temporary configuration

s0undt3ch committed:
Integration tests should not use the temporary config

s0undt3ch committed:
Use a temporary configuration instead

s0undt3ch committed:
Use a temporary configuration instead

s0undt3ch committed:
Use a temporary configuration instead

s0undt3ch committed:
Use a temporary configuration instead

s0undt3ch committed:
Use a temporary configuration instead

s0undt3ch committed:
Remove unused imports

s0undt3ch committed:
Ignore the unimplemented abstract method

s0undt3ch commented:
Go Go Jenkins!

s0undt3ch commented:
Go Go Jenkins!

s0undt3ch commented:
Cachim!

s0undt3ch commented:
@cachedout this is ready for a final review. No more commits will be added to this PR.

s0undt3ch commented:
Woohoo!!!! Gimme a few mins for another one like this :)
